ipa-core/src/protocol/prss/mod.rs

```diff
@@ -103,12 +61,25 @@ impl From<PrssIndex128> for u128 {
     }
 }
 
+impl TryFrom<u128> for PrssIndex128 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: u128) -> Result<Self, Self::Error> {
+        let value = u64::try_from(value)?;
+
+        Ok(Self {
+            index: u32::try_from(value >> 32).unwrap().into(),
```

Comment:
I don't understand why we are doing all of this fanciness here. Perhaps a comment could help explain the rationale.

Reply:
This code was moved from `From<u64>` implementation. The description of `PrssIndex128` is not included in the PR and that's why it probably looks confusing. Basically we need two indices to be able to generate more than 16 bytes of randomness: PRF use-case is one example where we need to get 256 bit values out of PRSS. AES cipher generates 16 byte blocks, so we need to call it more than once with unique IV if we want more entropy. 

Protocols request randomness using `RecordId` which is a 32 bit integer and we don't have a lot of bits there to ask for more than one value for the same index. So to keep the old code working, we just implemented 32 bit `offset` field  and used it to generate large chunks

* A protocol requests BA256 value for row 0, using `prss.generate(RecordId::from(0))`
* PRSS creates an index: `0:0` and generates the lowest 16 bytes of randomness
* PRSS creates index: `0:1` and generates the highest 16 bytes of randomness
* PRSS concatenates them and returns back to the protocol

Offsets are opaque to the MPC code and determined by the number of iterations required from PRSS to generate the desired value.